### PR TITLE
feat: add :reverse to cumulative functions

### LINKED ISSUE #771: Support :reverse option in cumulative functions
To match Explorer: https://github.com/elixir-nx/explorer/pull/200

### PR DESCRIPTION
**Pull request by polvalente**
closes #771

### CODE REVIEW

**Review by josevalim — APPROVED** [review]
Mix format and ship it!!!

File: nx/lib/nx.ex [hunk 1]
```diff
@@ -7183,6 +7184,26 @@ defmodule Nx do
...
+          [9, 11, 13],
+          [9, 12, 15]
+        ]
+      >
+
+      iex> Nx.cumulative_sum(Nx.iota({3, 3}), axis: 1, reverse: true)
+      #Nx.Tensor<
+        s64[3][3]
+        [
+          [2, 3, 3],
```
Comment (jonatanklosko) [inline nx/lib/nx.ex:7202]:
```elixir
iex> [0, 1, 2] |> Explorer.Series.from_list() |> Explorer.Series.cumulative_sum(reverse: true)
#Explorer.Series<
  integer[3]
  [3, 3, 2]
>
```

Reverse only determines the direction we go. The result is equivalent to `reverse + cumulative_* + reverse`.

The default windowed implementation could be optimised by padding right rather than left, but for torch I think we will need the double reverse.

Comment (jonatanklosko) [reply nx/lib/nx.ex:7202]: Neither numpy nor pytorch support this option, but [tf.math.cumsum](https://www.tensorflow.org/api_docs/python/tf/math/cumsum) does.

File: torchx/lib/torchx/backend.ex [hunk 2]
```diff
@@ -705,39 +705,107 @@ defmodule Torchx.Backend do
...
+    result = Torchx.cumulative_sum(t_tx, axis)
+
+    if reverse do
+      result
+      |> Torchx.flip([axis])
+      |> to_nx(out)
+    else
+      to_nx(result, out)
+    end
   end
```
Comment (jonatanklosko) [inline torchx/lib/torchx/backend.ex:732]: We can refactor it into `cumulative_op(out, t, opts, &Torchx.cumulative_sum/2)` :)

**Review by josevalim — APPROVED** [review]

File: nx/lib/nx.ex [hunk 3]
```diff
@@ -7194,6 +7215,7 @@ defmodule Nx do
   ## Options
 
     * `:axis` - the axis to multiply elements along. Defaults to `0`
+    * `:reverse` - if `true`, reverses the direction on the axis of accumulation. Defaults to `false`
```
Comment (jonatanklosko) [inline nx/lib/nx.ex:7218]:
```suggestion

    * `:reverse` - whether to perform accumulation in the opposite direction. Defaults to `false`
```

Comment (polvalente) [reply nx/lib/nx.ex:7218]: I don't like this because it doesn't lead the person to pass a boolean

Comment (jonatanklosko) [reply nx/lib/nx.ex:7218]: I'd say `` Defaults to `false` `` makes it clear, but your call!

File: nx/lib/nx.ex [hunk 4]
```diff
@@ -7324,7 +7411,20 @@ defmodule Nx do
...
+          tensor
+        end
+
+      result = aggregate_window_op(t, window_shape, [padding: padding], window_op)
+
+      if reverse do
+        Nx.reverse(result, axes: [axis])
+      else
+        result
+      end
```
[the remark this answers is not in the document]
Comment (polvalente) [reply nx/lib/nx.ex:7427]: This works beautifully!
It eliminates both reverses :)

File: nx/lib/nx.ex (hunk 3 above)
Comment (polvalente) [reply nx/lib/nx.ex:7218]: How about:

`* `:reverse` - a boolean which determines whether to perform accumulation in the opposite direction. Defaults to `false``

Comment (jonatanklosko) [reply nx/lib/nx.ex]: Works for me :) Sidenote, we currently have `` `:keep_axis` - whether or not ...``
